src/main/java/TypewiseAlert/AlertFactory.java

```diff
@@ -0,0 +1,9 @@
+package TypewiseAlert;
+
+public class AlertFactory {
+    public IAlertTarget AlertFactory(EnumConstants.AlertTarget alertTarget) {
+        if (alertTarget.equals(EnumConstants.AlertTarget.TO_CONTROLLER)) return new Controller();
+        else if (alertTarget.equals(EnumConstants.AlertTarget.TO_EMAIL)) return new EmailSender();
```

Comment:
Factory open for modification and closed for extensibility! Think about convention over configuration , hint: use reflection to find the call names who implements IAlertTarget